### PR TITLE
build for v6.1.0

### PR DESCRIPTION
**Pull request by calvinmetcalf**
seems like the big changes are the new buffer api, plus that fix of mine for throwing on null chunks

### CODE REVIEW

File: lib/_stream_readable.js [hunk 1]
```diff
@@ -149,7 +162,7 @@ Readable.prototype.push = function (chunk, encoding) {
   if (!state.objectMode && typeof chunk === 'string') {
     encoding = encoding || state.defaultEncoding;
     if (encoding !== state.encoding) {
-      chunk = new Buffer(chunk, encoding);
+      chunk = Buffer.from(chunk, encoding);
```
Comment (mafintosh) [inline lib/_stream_readable.js]: this doesn't work on older nodes right?

File: lib/_stream_readable.js [hunk 2]
```diff
@@ -825,16 +839,16 @@ function fromList(n, state) {
     } else {
       // complex case.
       // we have enough to cover it, but it spans past the first buffer.
-      if (stringMode) ret = '';else ret = new Buffer(n);
+      if (stringMode) ret = '';else ret = Buffer.allocUnsafe(n);
```
Comment (mafintosh) [inline lib/_stream_readable.js]: same as above

File: lib/_stream_writable.js [hunk 3]
```diff
@@ -265,11 +271,12 @@ Writable.prototype.setDefaultEncoding = function setDefaultEncoding(encoding) {
   if (typeof encoding === 'string') encoding = encoding.toLowerCase();
   if (!(['hex', 'utf8', 'utf-8', 'ascii', 'binary', 'base64', 'ucs2', 'ucs-2', 'utf16le', 'utf-16le', 'raw'].indexOf((encoding + '').toLowerCase()) > -1)) throw new TypeError('Unknown encoding: ' + encoding);
   this._writableState.defaultEncoding = encoding;
+  return this;
 };
 
 function decodeChunk(state, chunk, encoding) {
   if (!state.objectMode && state.decodeStrings !== false && typeof chunk === 'string') {
-    chunk = new Buffer(chunk, encoding);
+    chunk = Buffer.from(chunk, encoding);
```
Comment (mafintosh) [inline lib/_stream_writable.js]: same as above

File: lib/_stream_readable.js (hunk 1 above)
Comment (calvinmetcalf) [inline lib/_stream_readable.js]: yeah we're probably going to have to write a shim or polyfill

**Comment by calvinmetcalf** [comment]
next up will be a fix for deepStrictEqual

**Comment by calvinmetcalf** [comment]
defunctzombie/commonjs-assert#17

**Comment by Fishrock123** [comment]
@calvinmetcalf any progress? Need any help?

**Comment by calvinmetcalf** [comment]
I need to port deepStrictEqual to work on all the platforms defunctzombie/commonjs-assert#17 is not going as well as I'd hoped

**Comment by calvinmetcalf** [comment]
update looked like I found the problem, it was ArrayBuffer.isView

**Comment by calvinmetcalf** [comment]
and I should have been suspicious about all the file changes

**Comment by calvinmetcalf** [comment]
ok while we wait for @defunctzombie to review the changes to assert-commonjs we can test against my branch to make sure it actually fixes things

**Comment by defunctzombie** [comment]
@calvinmetcalf I've added you as collaborator to the commonjs-assert repo. I don't have much time for some of these projects these days and don't want to be a blocker.

**Comment by calvinmetcalf** [comment]
thanks!

**Comment by calvinmetcalf** [comment]
and got it working with defunctzombie/commonjs-assert#18

**Comment by calvinmetcalf** [comment]
ok I've had all browsers pass, just not in the same run so screw it calling that green and going to publish

**Pull request merged**